contracts/PassiveSwap/AbstractSwapper.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import { AccessControlledV8 } from "@venusprotocol/governance-contracts/contracts/Governance/AccessControlledV8.sol";
+import { ResilientOracle } from "@venusprotocol/oracle/contracts/ResilientOracle.sol";
+import { ensureNonzeroAddress } from "../lib/validators.sol";
+import { VTokenInterface } from "../VTokenInterfaces.sol";
+import { MANTISSA_ONE, EXP_SCALE } from "../lib/constants.sol";
+
+contract AbstractSwapper is AccessControlledV8 {
+    /// @notice Swap configuration for the tokens pair
+    struct SwapConfiguration {
+        /// tokenIn address
+        address tokenAddressIn;
+        /// tokenOut address
+        address tokenAddressOut;
+        /// incentive on swapping tokens
+        uint256 incentive;
+        /// whether the swap is enabled
+        bool enabled;
+    }
+
+    ResilientOracle public priceOracle;
+
```

Comment:
This contract will be inherited in child contract for that reason we should add a `--gap[]` variable here

Reply:
Added, d1c6ee3894085fd33e8ac98110d52004eab99ac9